// @airwallex/node-sdk/dist/api/paymentAcceptance/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAcceptance = void 0;
const autoPaymentsApi_1 = require("./autoPaymentsApi");
const paymentAcceptanceConfigApi_1 = require("./paymentAcceptanceConfigApi");
const paymentAcceptanceCustomersApi_1 = require("./paymentAcceptanceCustomersApi");
const paymentAcceptanceCustomsDeclarationsApi_1 = require("./paymentAcceptanceCustomsDeclarationsApi");
const paymentAcceptanceFundsSplitDestinationsApi_1 = require("./paymentAcceptanceFundsSplitDestinationsApi");
const paymentAcceptanceFundsSplitReversalsApi_1 = require("./paymentAcceptanceFundsSplitReversalsApi");
const paymentAcceptanceFundsSplitsApi_1 = require("./paymentAcceptanceFundsSplitsApi");
const paymentAcceptancePaymentAttemptsApi_1 = require("./paymentAcceptancePaymentAttemptsApi");
const paymentAcceptancePaymentConsentsApi_1 = require("./paymentAcceptancePaymentConsentsApi");
const paymentAcceptancePaymentDisputesApi_1 = require("./paymentAcceptancePaymentDisputesApi");
const paymentAcceptancePaymentIntentsApi_1 = require("./paymentAcceptancePaymentIntentsApi");
const paymentAcceptancePaymentMethodsApi_1 = require("./paymentAcceptancePaymentMethodsApi");
const paymentAcceptanceRefundsApi_1 = require("./paymentAcceptanceRefundsApi");
const paymentAcceptanceSettlementRecordsApi_1 = require("./paymentAcceptanceSettlementRecordsApi");
const paymentAcceptanceTerminalsApi_1 = require("./paymentAcceptanceTerminalsApi");
const reservesApi_1 = require("./reservesApi");
class PaymentAcceptance {
    constructor() {
        this.autoPayments = new autoPaymentsApi_1.AutoPaymentsApi();
        this.config = new paymentAcceptanceConfigApi_1.PaymentAcceptanceConfigApi();
        this.customers = new paymentAcceptanceCustomersApi_1.PaymentAcceptanceCustomersApi();
        this.customsDeclarations = new paymentAcceptanceCustomsDeclarationsApi_1.PaymentAcceptanceCustomsDeclarationsApi();
        this.fundsSplitDestinations = new paymentAcceptanceFundsSplitDestinationsApi_1.PaymentAcceptanceFundsSplitDestinationsApi();
        this.fundsSplitReversals = new paymentAcceptanceFundsSplitReversalsApi_1.PaymentAcceptanceFundsSplitReversalsApi();
        this.fundsSplits = new paymentAcceptanceFundsSplitsApi_1.PaymentAcceptanceFundsSplitsApi();
        this.paymentAttempts = new paymentAcceptancePaymentAttemptsApi_1.PaymentAcceptancePaymentAttemptsApi();
        this.paymentConsents = new paymentAcceptancePaymentConsentsApi_1.PaymentAcceptancePaymentConsentsApi();
        this.paymentDisputes = new paymentAcceptancePaymentDisputesApi_1.PaymentAcceptancePaymentDisputesApi();
        this.paymentIntents = new paymentAcceptancePaymentIntentsApi_1.PaymentAcceptancePaymentIntentsApi();
        this.paymentMethods = new paymentAcceptancePaymentMethodsApi_1.PaymentAcceptancePaymentMethodsApi();
        this.refunds = new paymentAcceptanceRefundsApi_1.PaymentAcceptanceRefundsApi();
        this.settlementRecords = new paymentAcceptanceSettlementRecordsApi_1.PaymentAcceptanceSettlementRecordsApi();
        this.terminals = new paymentAcceptanceTerminalsApi_1.PaymentAcceptanceTerminalsApi();
        this.reserves = new reservesApi_1.ReservesApi();
    }
}
exports.PaymentAcceptance = PaymentAcceptance;
