// @airwallex/node-sdk/dist/api/paymentAcceptance/paymentAcceptancePaymentAttemptsApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAcceptancePaymentAttemptsApi = void 0;
const apiClient_1 = require("../../client/apiClient");
const dataParser_1 = require("../../utils/dataParser");
const paymentAcceptanceListResponseOfPaymentAcceptancePaymentAttempt_1 = require("../../model/paymentAcceptanceListResponseOfPaymentAcceptancePaymentAttempt");
const paymentAcceptancePaymentAttempt_1 = require("../../model/paymentAcceptancePaymentAttempt");
class PaymentAcceptancePaymentAttemptsApi {
    async listPaymentAttempts({ currency, from_created_at, page_num, page_size, payment_intent_id, status, to_created_at }, options = {}) {
        const localVarPath = '/api/v1/pa/payment_attempts';
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        let queryParams = {};
        if (currency !== undefined) {
            queryParams['currency'] = currency;
        }
        if (from_created_at !== undefined) {
            queryParams['from_created_at'] = from_created_at;
        }
        if (page_num !== undefined) {
            queryParams['page_num'] = page_num;
        }
        if (page_size !== undefined) {
            queryParams['page_size'] = page_size;
        }
        if (payment_intent_id !== undefined) {
            queryParams['payment_intent_id'] = payment_intent_id;
        }
        if (status !== undefined) {
            queryParams['status'] = status;
        }
        if (to_created_at !== undefined) {
            queryParams['to_created_at'] = to_created_at;
        }
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            params: queryParams,
            timeout,
            responseType,
        });
        const typeName = paymentAcceptanceListResponseOfPaymentAcceptancePaymentAttempt_1.PaymentAcceptanceListResponseOfPaymentAcceptancePaymentAttempt.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
    async retrievePaymentAttempt(id, options = {}) {
        if (id === null || id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling retrievePaymentAttempt.');
        }
        const localVarPath = '/api/v1/pa/payment_attempts/{id}'.replace('{' + 'id' + '}', encodeURIComponent(String(id)));
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            timeout,
            responseType,
        });
        const typeName = paymentAcceptancePaymentAttempt_1.PaymentAcceptancePaymentAttempt.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
}
exports.PaymentAcceptancePaymentAttemptsApi = PaymentAcceptancePaymentAttemptsApi;
