// @airwallex/node-sdk/dist/api/paymentAcceptance/paymentAcceptanceSettlementRecordsApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAcceptanceSettlementRecordsApi = void 0;
const apiClient_1 = require("../../client/apiClient");
const dataParser_1 = require("../../utils/dataParser");
const paymentAcceptanceForwardNavigationListResponseOfPaymentAcceptanceSettlementRecord_1 = require("../../model/paymentAcceptanceForwardNavigationListResponseOfPaymentAcceptanceSettlementRecord");
class PaymentAcceptanceSettlementRecordsApi {
    async listSettlementRecords({ page, page_size, payment_intent_id, settlement_id }, options = {}) {
        const localVarPath = '/api/v1/pa/settlement_records';
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        let queryParams = {};
        if (page !== undefined) {
            queryParams['page'] = page;
        }
        if (page_size !== undefined) {
            queryParams['page_size'] = page_size;
        }
        if (payment_intent_id !== undefined) {
            queryParams['payment_intent_id'] = payment_intent_id;
        }
        if (settlement_id !== undefined) {
            queryParams['settlement_id'] = settlement_id;
        }
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            params: queryParams,
            timeout,
            responseType,
        });
        const typeName = paymentAcceptanceForwardNavigationListResponseOfPaymentAcceptanceSettlementRecord_1.PaymentAcceptanceForwardNavigationListResponseOfPaymentAcceptanceSettlementRecord.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
}
exports.PaymentAcceptanceSettlementRecordsApi = PaymentAcceptanceSettlementRecordsApi;
