// @airwallex/node-sdk/dist/api/paymentAcceptance/paymentAcceptanceFundsSplitsApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAcceptanceFundsSplitsApi = void 0;
const apiClient_1 = require("../../client/apiClient");
const dataParser_1 = require("../../utils/dataParser");
const paymentAcceptanceFundSplit_1 = require("../../model/paymentAcceptanceFundSplit");
const paymentAcceptanceListResponseOfPaymentAcceptanceFundSplit_1 = require("../../model/paymentAcceptanceListResponseOfPaymentAcceptanceFundSplit");
class PaymentAcceptanceFundsSplitsApi {
    async create(funds_split_create_request, x_on_behalf_of, options = {}) {
        const localVarPath = '/api/v1/pa/funds_splits/create';
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        if (funds_split_create_request === null || funds_split_create_request === undefined) {
            throw new Error('Required parameter funds_split_create_request was null or undefined when calling create.');
        }
        localVarHeaderParams['x-on-behalf-of'] = x_on_behalf_of;
        const data = await apiClient_1.apiClient.post(localVarPath, {
            headers: localVarHeaderParams,
            timeout,
            responseType,
        }, funds_split_create_request);
        const typeName = paymentAcceptanceFundSplit_1.PaymentAcceptanceFundSplit.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
    async list({ page_num, page_size, source_id, source_type }, options = {}) {
        const localVarPath = '/api/v1/pa/funds_splits';
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        let queryParams = {};
        if (page_num !== undefined) {
            queryParams['page_num'] = page_num;
        }
        if (page_size !== undefined) {
            queryParams['page_size'] = page_size;
        }
        if (source_id !== undefined) {
            queryParams['source_id'] = source_id;
        }
        if (source_type !== undefined) {
            queryParams['source_type'] = source_type;
        }
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            params: queryParams,
            timeout,
            responseType,
        });
        const typeName = paymentAcceptanceListResponseOfPaymentAcceptanceFundSplit_1.PaymentAcceptanceListResponseOfPaymentAcceptanceFundSplit.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
    async release(id, FundSplit_Release_Request, options = {}) {
        if (id === null || id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling release.');
        }
        const localVarPath = '/api/v1/pa/funds_splits/{id}/release'.replace('{' + 'id' + '}', encodeURIComponent(String(id)));
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        if (FundSplit_Release_Request === null || FundSplit_Release_Request === undefined) {
            throw new Error('Required parameter FundSplit_Release_Request was null or undefined when calling release.');
        }
        const data = await apiClient_1.apiClient.post(localVarPath, {
            headers: localVarHeaderParams,
            timeout,
            responseType,
        }, FundSplit_Release_Request);
        const typeName = paymentAcceptanceFundSplit_1.PaymentAcceptanceFundSplit.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
    async retrieveFundsSplit(id, options = {}) {
        if (id === null || id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling retrieveFundsSplit.');
        }
        const localVarPath = '/api/v1/pa/funds_splits/{id}'.replace('{' + 'id' + '}', encodeURIComponent(String(id)));
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            timeout,
            responseType,
        });
        const typeName = paymentAcceptanceFundSplit_1.PaymentAcceptanceFundSplit.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
}
exports.PaymentAcceptanceFundsSplitsApi = PaymentAcceptanceFundsSplitsApi;
