// @airwallex/node-sdk/dist/api/simulation/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Simulation = void 0;
const simulationSandboxOnlyAccountsApi_1 = require("./simulationSandboxOnlyAccountsApi");
const simulationSandboxOnlyDepositsApi_1 = require("./simulationSandboxOnlyDepositsApi");
const simulationSandboxOnlyIssuingApi_1 = require("./simulationSandboxOnlyIssuingApi");
const simulationSandboxOnlyLinkedAccountsApi_1 = require("./simulationSandboxOnlyLinkedAccountsApi");
const simulationSandboxOnlyPaymentAcceptanceApi_1 = require("./simulationSandboxOnlyPaymentAcceptanceApi");
const simulationSandboxOnlyPayoutsApi_1 = require("./simulationSandboxOnlyPayoutsApi");
const simulationSandboxOnlyRequestForInformationRFIApi_1 = require("./simulationSandboxOnlyRequestForInformationRFIApi");
const simulationSandboxOnlyTransfersApi_1 = require("./simulationSandboxOnlyTransfersApi");
class Simulation {
    constructor() {
        this.accounts = new simulationSandboxOnlyAccountsApi_1.SimulationSandboxOnlyAccountsApi();
        this.deposits = new simulationSandboxOnlyDepositsApi_1.SimulationSandboxOnlyDepositsApi();
        this.issuing = new simulationSandboxOnlyIssuingApi_1.SimulationSandboxOnlyIssuingApi();
        this.linkedAccounts = new simulationSandboxOnlyLinkedAccountsApi_1.SimulationSandboxOnlyLinkedAccountsApi();
        this.paymentAcceptance = new simulationSandboxOnlyPaymentAcceptanceApi_1.SimulationSandboxOnlyPaymentAcceptanceApi();
        this.payouts = new simulationSandboxOnlyPayoutsApi_1.SimulationSandboxOnlyPayoutsApi();
        this.requestForInformationRFI = new simulationSandboxOnlyRequestForInformationRFIApi_1.SimulationSandboxOnlyRequestForInformationRFIApi();
        this.transfers = new simulationSandboxOnlyTransfersApi_1.SimulationSandboxOnlyTransfersApi();
    }
}
exports.Simulation = Simulation;
