// @airwallex/node-sdk/dist/api/transactionalFX/transactionalFXRatesApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionalFXRatesApi = void 0;
const apiClient_1 = require("../../client/apiClient");
const dataParser_1 = require("../../utils/dataParser");
const fxRatesDto_1 = require("../../model/fxRatesDto");
class TransactionalFXRatesApi {
    async getCurrentFxRate({ buy_amount, buy_currency, conversion_date, sell_amount, sell_currency }, options = {}) {
        const localVarPath = '/api/v1/fx/rates/current';
        const { headers, timeout, responseType } = options;
        let localVarHeaderParams = Object.assign({}, headers ? headers : {});
        let queryParams = {};
        if (buy_amount !== undefined) {
            queryParams['buy_amount'] = buy_amount;
        }
        if (buy_currency !== undefined) {
            queryParams['buy_currency'] = buy_currency;
        }
        if (conversion_date !== undefined) {
            queryParams['conversion_date'] = conversion_date;
        }
        if (sell_amount !== undefined) {
            queryParams['sell_amount'] = sell_amount;
        }
        if (sell_currency !== undefined) {
            queryParams['sell_currency'] = sell_currency;
        }
        const data = await apiClient_1.apiClient.get(localVarPath, {
            headers: localVarHeaderParams,
            params: queryParams,
            timeout,
            responseType,
        });
        const typeName = fxRatesDto_1.FxRatesDto.discriminator;
        return (0, dataParser_1.parseDataToType)(data, typeName);
    }
}
exports.TransactionalFXRatesApi = TransactionalFXRatesApi;
