// @airwallex/node-sdk/dist/api/issuing/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Issuing = void 0;
const issuingAuthorizationsApi_1 = require("./issuingAuthorizationsApi");
const issuingCardholdersApi_1 = require("./issuingCardholdersApi");
const issuingCardsApi_1 = require("./issuingCardsApi");
const issuingConfigApi_1 = require("./issuingConfigApi");
const issuingDigitalWalletTokensApi_1 = require("./issuingDigitalWalletTokensApi");
const issuingTransactionDisputesApi_1 = require("./issuingTransactionDisputesApi");
const issuingTransactionLifecycleEventsApi_1 = require("./issuingTransactionLifecycleEventsApi");
const issuingTransactionLifecyclesApi_1 = require("./issuingTransactionLifecyclesApi");
const issuingTransactionsApi_1 = require("./issuingTransactionsApi");
class Issuing {
    constructor() {
        this.authorizations = new issuingAuthorizationsApi_1.IssuingAuthorizationsApi();
        this.cardholders = new issuingCardholdersApi_1.IssuingCardholdersApi();
        this.cards = new issuingCardsApi_1.IssuingCardsApi();
        this.config = new issuingConfigApi_1.IssuingConfigApi();
        this.digitalWalletTokens = new issuingDigitalWalletTokensApi_1.IssuingDigitalWalletTokensApi();
        this.transactionDisputes = new issuingTransactionDisputesApi_1.IssuingTransactionDisputesApi();
        this.transactionLifecycleEvents = new issuingTransactionLifecycleEventsApi_1.IssuingTransactionLifecycleEventsApi();
        this.transactionLifecycles = new issuingTransactionLifecyclesApi_1.IssuingTransactionLifecyclesApi();
        this.transactions = new issuingTransactionsApi_1.IssuingTransactionsApi();
    }
}
exports.Issuing = Issuing;
